rust/lance/src/index/prefilter.rs

```diff
@@ -62,7 +62,7 @@ impl DatasetPreFilter {
     ) -> Self {
         let mut fragments = RoaringBitmap::new();
         if indices.iter().any(|idx| idx.fragment_bitmap.is_none()) {
-            fragments.insert_range(0..dataset.manifest.max_fragment_id);
+            fragments.insert_range(0..dataset.manifest.max_fragment_id.unwrap_or(0));
```

Comment:
It looks like we unwrap to 0 in all cases where it has never been set (e.g. returns None).  Do we want to move this logic into `max_fragment_id`?

Reply:
In all of these cases it makes sense. It might not universally make sense. It's meant to differentiate between:

```
fragments: []
```

and 
```
fragments: [Fragment(id=0)]
```

So want to keep that distinction